src/docGenerator.ts

```diff
@@ -0,0 +1,146 @@
+import fs from "fs"
+import { JSONSchema4 } from "json-schema"
+import { flatMap, flatMapDeep, partition, toPairs } from "lodash"
+import fetch from "node-fetch"
+import { promisify } from "util"
+import { defaultEngine } from "./eslintDefaultOptions"
+import { DescriptionEntry } from "./model/Description"
+import {
+  Category,
+  fromEslintCategoryToCategory,
+  fromEslintCategoryToLevel,
+  Level,
+  patternIdToCodacy,
+  Patterns,
+  PatternsEntry,
+  PatternsParameter
+} from "./model/Patterns"
+import { toolName, toolVersion } from "./toolMetadata"
+
+const writeFile = promisify(fs.writeFile)
+
+export function generatePatterns(): Patterns {
+  const rules = defaultEngine.getRules()
```

Comment:
shouldn't `rules` come as an argument to this function?

Reply:
Done ✅ 
I made rules field of the DocGenerator class